### ISSUE TITLE
With `#[context]`, a function never gets marked unused

### ISSUE THREAD

**Issue by jplatte** [andrewhickman/fn-error-context#3]
Title should be self-explanatory. This should result in an unused function warning:

```rust
#[context("foo")]
fn never_used() -> anyhow::Result<()> {
    Ok(())
}
```

**Comment by andrewhickman** [comment]
Unfortunately this looks like a limitation of rust macros. All lints originating from inside an external macro are suppressed (https://github.com/rust-lang/rust/pull/52467)

**Comment by andrewhickman** [comment]
After some experimentation, it turns out we were losing the spans on the `{ }` tokens in the macro output. If we pass them them through directly, rustc seems to decide they're part of the original crate for the purposes of linting

Thanks for the issue!

**Comment by andrewhickman** [comment]
I've published [0.1.2](https://crates.io/crates/fn-error-context/0.1.2) with the fix for this
